src/processor/dataStack.ts

```diff
@@ -69,15 +69,31 @@ export default function dataStack(ecModel: GlobalModel) {
                 return;
             }
 
-            stackInfoList.length && data.setCalculationInfo(
-                'stackedOnSeries', stackInfoList[stackInfoList.length - 1].seriesModel
-            );
-
             stackInfoList.push(stackInfo);
         }
     });
 
-    stackInfoMap.each(calculateStack);
+    // Process each stack group
+    stackInfoMap.each(function (stackInfoList) {
+        // Check if stack order needs to be reversed
+        const firstSeries = stackInfoList[0].seriesModel;
```

Comment:
Although `stackInfoList.length` is not expected to be 0 here, it's safer to add `if (stackInfoList.length === 0) { return; }` to align with the original logic.

Reply:
done